### PR TITLE
Detect icmpv6hdr v7

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3323 

Describe changes:
- Fixes some typos
- Adds a generic `DetectU32Data` structure for signature matches on uint32 values
- Adds `icmpv6.mtu` signature  keyword
- Fix endianness for ICMPv6 MTU decoding

This fixes only the first evasion of the redmine ticket

Makes https://github.com/OISF/suricata-verify/pull/187 pass

Modifies #4639 by taking comments into account

There is still some points to improve/discuss :
- Handle `DetectBufferTypeRegister` as fatal error... Should it be part of this PR ?
- Use `pcre_copy_substring` in detect-tcpmss.c... Should it be part of this PR ?

### CODE REVIEW

File: src/detect-engine-prefilter-common.h [hunk 1]
```diff
@@ -94,4 +94,14 @@ PrefilterPacketHeaderExtraMatch(const PrefilterPacketHeaderCtx *ctx,
     return TRUE;
 }
 
+static inline bool PrefilterIsPrefilterableById(const Signature *s, enum DetectKeywordId kid) {
```
Comment (victorjulien) [inline src/detect-engine-prefilter-common.h]: Please add this to its own commit as well.

File: src/detect-icmpv6-mtu.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+     * check that can be added: alproto, sport or dport */
+    const PrefilterPacketHeaderCtx *ctx = pectx;
+    if (PrefilterPacketHeaderExtraMatch(ctx, p) == FALSE)
+        return;
+
+    /* if we match, add all the sigs that use this prefilter. This means
+     * that these will be inspected further */
+    DetectU32Data du32;
+    du32.mode = ctx->v1.u8[0];
+    du32.arg1 = ctx->v1.u32[1];
```
Comment (victorjulien) [inline src/detect-icmpv6-mtu.c]: Should u32[2] be included here as well?

Comment (catenacyber) [reply src/detect-icmpv6-mtu.c]: Well seen, thanks :-)

File: src/detect-engine-prefilter-common.h (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-prefilter-common.h]: ok

**Pull request closed without merging**